docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,165 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+With the User Agent Filter module, you can allow or deny traffic through regular expression-based rules that are matched against the incoming `User-Agent` header. The `User-Agent` header contains information about the requesting browser or application, including its name, version, and operating system.
+
+Since rules are processed at the ngrok Cloud Edge, you can bolster your application’s security by blocking unwanted or unauthorized requests from known bots and malicious actors before it ever reaches your network.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 \
+  --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" \
+  --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+  example:
+    proto: http
+    addr: 80
+    ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+    ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
```

Comment:
Can users use the `WithDeny` / `WithAllow` for go as well? I think we should also link the sdks from here.

> Here is an example using the [ngrok-library](link)